microservices-connector/internal/controller/gmc_monitor.go

```diff
@@ -0,0 +1,159 @@
+/*
+* Copyright (C) 2024 Intel Corporation
+* SPDX-License-Identifier: Apache-2.0
+ */
+
+package controller
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	mcv1alpha3 "github.com/opea-project/GenAIInfra/microservices-connector/api/v1alpha3"
+	appsv1 "k8s.io/api/apps/v1"
+	corev1 "k8s.io/api/core/v1"
+	apierr "k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+type GMC_monitor struct {
+	client.Client
+	Scheme *runtime.Scheme
+
+	// ResourceStatus is a map of service name to its status
+	ResourceStatus map[NsName](MonitorCategory)
+}
+type MonitorCategory struct {
+	Graph         NsName
+	resources     []ServiceAndDeploy
+	readyCount    int
+	externalCount int
+	totalCount    int
+}
+
+type NsName struct {
+	Namespace string
+	Name      string
+}
+
+type ServiceAndDeploy struct {
+	srvc NsName
+	dply NsName
+}
+
+func (m GMC_monitor) Start(ch <-chan MonitorCategory, stopCh <-chan struct{}) {
+	go func() {
+		fmt.Println("Monitor task initiated.")
+
+		for {
+			select {
+			case mc := <-ch:
+				fmt.Printf(" received monitor object: %v \n", mc)
+				// Process the MonitorCategory
+				m.ResourceStatus[mc.Graph] = mc
+			case <-stopCh:
+				// Exit the goroutine when stop signal is received
+				return
+			}
+
+		}
+	}()
+
+	// Start polling in a separate goroutine
+	go func() {
+		ticker := time.NewTicker(30 * time.Second) // Adjust the interval as needed
```

Comment:
is it reasonable to set to 30 seconds for interval?

Reply:
do you have any suggestions? I'd say we can have a trial on this value, if not meeting any requirements, then we make the adjustments.